src/Library/demos/Text View/main.vala

```diff
@@ -0,0 +1,76 @@
+    -1
+  );
+
+  copy.clicked.connect(handleCopy);
+  paste.clicked.connect(handlePaste);
+  cut.clicked.connect(handleCut);
+  select.clicked.connect(() => {
+    Gtk.TextIter start, end;
+
+    buffer.get_start_iter(out start);
+    buffer.get_end_iter(out end);
+
+    buffer.select_range(start, end);
+  });
+  clear.clicked.connect(() => {
+    Gtk.TextIter start, end;
+
+    buffer.get_start_iter(out start);
+    buffer.get_end_iter(out end);
+
+    buffer.@delete(ref start, ref end);
+  });
+
+  textview.copy_clipboard.connect(() => {
+    message("Text copied to clipboard");
+  });
+  textview.cut_clipboard.connect(() => {
+    message("Text cut to clipboard");
+  });
+  textview.paste_clipboard.connect(() => {
+    message("Text pasted");
+  });
+}
+
+void handleCopy() {
+  if (buffer.get_has_selection()) {
+    buffer.copy_clipboard(clipboard);
```

Comment:
Always use properties if they are available. In this case `if (buffer.has_selection) {`